rust/operator-binary/src/service.rs

```diff
@@ -104,7 +105,7 @@ fn headless_service_ports() -> Vec<ServicePort> {
     }]
 }
 
-fn metrics_service_ports() -> Vec<ServicePort> {
+pub fn metrics_service_ports() -> Vec<ServicePort> {
```

Comment:
nit: I would prefer renaming it to `metrics_service_port`. I was shortly confused why there are multiple metric ports.

Regarding my other comment: We could make it

`pub fn metrics_service_port(product_version: &str) -> ServicePort {`
and let it do the v1/v2 switching. That way we have a clear name and purpose of the fn as well as rustdocs

Reply:
https://github.com/stackabletech/nifi-operator/pull/819/commits/530f16c61c367ca937b64cc7efaa8e538f1d9eeb